mrp_work_center_component/models/mrp_production.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2025 Angel Rivas <angel.rivas@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        productions = super().create(vals_list)
+        productions._add_workcenter_components()
+        return productions
+
+    def _add_workcenter_components(self):
+        StockMove = self.env["stock.move"]
+        WorkcenterComponent = self.env["mrp.workcenter.component"]
+
+        for production in self:
+            components = WorkcenterComponent.search(
+                [
+                    (
+                        "work_center_id",
+                        "in",
+                        production.workorder_ids.mapped("workcenter_id").ids,
+                    )
+                ]
+            )
+            move_vals = []
+            for component in components:
+                move_vals.append(
```

Comment:
Extract this section into a function. As is done for data from creating delivery notes from sales, etc.
If you do it this way it will be very difficult to carry out an inheritance if necessary.